android-components/components/feature/media/src/main/java/mozilla/components/feature/media/fullscreen/MediaSessionFullscreenFeature.kt

```diff
@@ -35,51 +35,53 @@ class MediaSessionFullscreenFeature(
             flow.map {
                 it.tabs + it.customTabs
             }.map { tab ->
-                tab.filter { it.mediaSessionState != null && it.mediaSessionState!!.fullscreen }
-            }.distinctUntilChanged().collect { states ->
-                processFullscreen(states)
-                processDeviceSleepMode(states)
+                tab.firstOrNull { it.mediaSessionState != null && it.mediaSessionState!!.fullscreen }
+            }.distinctUntilChanged { old, new ->
+                old?.mediaSessionState?.fullscreen == new?.mediaSessionState?.fullscreen &&
+                    old?.mediaSessionState?.playbackState == new?.mediaSessionState?.playbackState &&
+                    old?.content?.pictureInPictureEnabled == new?.content?.pictureInPictureEnabled &&
+                    old?.mediaSessionState?.elementMetadata == new?.mediaSessionState?.elementMetadata
+            }.collect { state -> /* there should only be one fullscreen session */
+                if (state == null || state.mediaSessionState?.fullscreen != true) {
+                    activity.requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_USER
+                    activity.window?.clearFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON)
+                    return@collect
+                }
+
+                if (store.state.findCustomTabOrSelectedTab(tabId)?.id == state.id) {
+                    processFullscreen(state)
+                }
+                processDeviceSleepMode(state)
             }
         }
     }
 
     @Suppress("SourceLockedOrientationActivity") // We deliberately want to lock the orientation here.
-    private fun processFullscreen(sessionStates: List<SessionState>) {
-        /* there should only be one fullscreen session */
-        val activeState = sessionStates.firstOrNull()
-        if (activeState == null || activeState.mediaSessionState?.fullscreen != true) {
-            activity.requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_USER
-            return
-        }
+    private fun processFullscreen(activeTabState: SessionState) {
```

Comment:
Also while we're here, would something like `setFullscreenOrientation` make more sense as a function name?